Tests/MapboxNavigationTests/VanishingRouteLineTests.swift

```diff
@@ -0,0 +1,262 @@
+        routeProgress.currentLegProgress.currentStepProgress = RouteStepProgress(step: route.legs[0].steps[2], spokenInstructionIndex: 0)
+        return routeProgress
+    }
+    
+    func lineGradientToString(lineGradient: Value<StyleColor>?) -> String {
+        guard let halfStringFromLineGradient = lineGradient.debugDescription.components(separatedBy: "(").last,
+              let stringFromLineGradient = halfStringFromLineGradient.components(separatedBy: ")").first else {
+            preconditionFailure("Line gradient string is invalid.")
+        }
+        return stringFromLineGradient
+    }
+    
+    func setUpCameraZoom(at zoomeLevel: CGFloat) {
+        let cameraState = navigationMapView.mapView.cameraState
+        let cameraOption = CameraOptions(center: cameraState.center, padding: cameraState.padding, zoom: zoomeLevel, bearing: cameraState.bearing, pitch: cameraState.pitch)
+        navigationMapView.mapView.camera.ease(to: cameraOption, duration: 0.1, curve: .linear)
+        
+        expectation(description: "Zoom set up") {
+            self.navigationMapView.mapView.cameraState.zoom == zoomeLevel
+        }
+        waitForExpectations(timeout: 2, handler: nil)
+    }
+    
+    func testParseRoutePoints() {
+        // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtilsTest.kt#L1798
+        let route = getMultilegRoute()
+        let routePoints = navigationMapView.parseRoutePoints(route: route)
+        
+        // Because mapbox-directions-swift parses the route with one more duplicate coordinate in the last step of each route leg. The two leg route has two more coordinates compared with Android.
+        XCTAssertEqual(routePoints.flatList.count, 130)
+        XCTAssertEqual(routePoints.nestedList.flatMap{$0}.count, 15)
+        XCTAssertEqual(routePoints.flatList[1].latitude, routePoints.flatList[2].latitude)
+        XCTAssertEqual(routePoints.flatList[1].longitude, routePoints.flatList[2].longitude)
+        XCTAssertEqual(routePoints.flatList[128].latitude, routePoints.flatList[129].latitude, accuracy: 0.000001)
+        XCTAssertEqual(routePoints.flatList[128].longitude, routePoints.flatList[129].longitude, accuracy: 0.000001)
+    }
+    
+    func testUpdateUpcomingRoutePointIndex() {
+        // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApiTest.kt#L802
+        let route = getRoute()
+        
+        navigationMapView.initPrimaryRoutePoints(route: route)
+        navigationMapView.routeLineGranularDistances = nil
+        XCTAssertEqual(navigationMapView.fractionTraveled, 0.0)
+        
+        let routeProgress = getRouteProgress()
```

Comment:
It is unclear what this method returns, can you please improve its signature or provide documentation comment? E.g. why stepIndex is set to `2` specifically etc.

Reply:
Thanks for the suggestion, comments are added in each unit test for documentation.